functions/graphql/index.js

```diff
@@ -24,14 +25,54 @@ function configureServer() {
   //use the cors middleware
   app.use(cors());
 
+  // Set up for the file middleware
+  app.use((req, res, next) => {
+    if (req.body instanceof Buffer) {
+      req.body = {};
+    }
+    next();
+  });
+
+  app.use(fileMiddleware);
```

Comment:
So does this mean that when Apollo gets a request, it first passes the request and response objects to this middleware stuff? I'm not really sure what the code right after this is doing , with the if req.files {} stuff. Would you mind giving me a breif break-down of what the parts here are doing and why?

Reply:
Yeah, there's a lot going on. Here's the gist:

Firebase functions do funky things to file uploads. There is file upload support built into Apollo Server, but it was expecting the upload bundle to be in a slightly different shape. As such, when firebase messed with it, Apollo Server freaked out and couldn't handle it.

The file middleware processes the request and turns it into something that can be handled by Apollo Server. It pulls out the file and puts it into `req.files` and pulls out the actual GraphQL request and puts it into `req.body`.

The next function has to map the files onto the variables. Since my GraphQL request uses a file upload as a variable, I have to take that file and put it into the correct variable so the GraphQL Resolvers can handle it. When the file is uploaded, it provides data for the proper mapping for the file to the variable. The middleware below the `app.use(fileMiddleware)` takes that mapping and puts it into the variables.

Does that make sense?